src/products/views/ProductCreate/ProductCreate.tsx

```diff
@@ -133,17 +133,32 @@ export const ProductCreateView: React.FC = () => {
     }
   });
 
-  const handleSubmit = createMetadataCreateHandler(
-    createHandler(
-      productTypes,
-      variables => productCreate({ variables }),
-      variables => productVariantCreate({ variables }),
-      updateChannels,
-      updateVariantChannels
-    ),
-    updateMetadata,
-    updatePrivateMetadata
-  );
+  const handleSubmit = async data => {
+    const result = await createMetadataCreateHandler(
+      createHandler(
+        productTypes,
+        variables => productCreate({ variables }),
+        variables => productVariantCreate({ variables }),
+        updateChannels,
+        updateVariantChannels,
+        deleteProduct
+      ),
+      updateMetadata,
+      updatePrivateMetadata
+    )(data);
+
+    if (result) {
+      setProductCreateComplete(true);
+    }
+  };
+
+  React.useEffect(() => {
+    const productId = productCreateOpts.data?.productCreate?.product?.id;
+
+    if (productCreateComplete && productId) {
+      handleSuccess(productId);
+    }
+  }, [productCreateComplete]);
```

Comment:
I think we can add this to handler too and skip `productCreateComplete` state from component 🤷‍♂️